src/material/core/typography/_typography.scss

```diff
@@ -69,6 +69,37 @@
   @return map-merge($config, (font-family: $font-family));
 }
 
+// Whether a config is for the Material Design 2018 typography system.
+@function mat-private-typography-is-2018-config($config) {
+  @if map-get($config, headline-1) {
+    @return true;
+  }
+  @return false;
+}
+
+// Given a config for either the 2014 or 2018 Material Design typography system,
+// produces a normalized typography config for the 2014 Material Design typography system.
+// 2014 - https://material.io/archive/guidelines/style/typography.html#typography-styles
+// 2018 - https://material.io/design/typography/the-type-system.html#type-scale
+@function mat-typography-normalized-config($config) {
```

Comment:
Could we bake this into `mat-typography` config and have it convert automatically. We could only do it if we can determine with some degree of certainty it's an actual 2018 config.

Reply:
We need this to be separate as we plan to use it to repair broken configs that are accidentally generated by some code inside Google.

Reply:
My hope was that we'd be able to slowly transition external apps to the 2018 config, but that can always be done later on.